### DISCUSSION TITLE
Get SharePoint Online usage data

### DISCUSSION THREAD

**Discussion by Oregonfrogger** [pnp/powershell#3980] [General]
Hello,

I'm new here to this group and I have seen some thoughts on getting usage data from SharePoint Online.  However, I have not gotten any to work, I think because Microsoft keeps changing things.  What I'm looking for specifically is the following:  Last activity, created, files viewed, page views, page visits, and files.  Is there any discussion on creating something for PnP that will retrieve this information? 

Thanks,

Darin

**Comment by Barbarur** [comment]
That seems a very general question. Not sure if you are looking for a script to retrieve that data?

Last activity and created are fields coming on all the objects (SiteProperties, List, ListItem).

For the Page analytics, you can try Get-PnPSiteAnaliticsData.

**Comment by Oregonfrogger** [comment]
Sorry about the general question.

I’m looking for the information from the site properties with a script to retrieve the data.  When I’m on the SharePoint Admin Center, the main view for active sites has the information that is desired.  Also if I select one of the sites then select Activity I will get that same information.  What I’m looking for is to have a script run daily and populate a list that can be viewed by others in the organization that don’t have access to the admin center.  I could go to the admin center and export the information into a csv, but I would like to do this all in a script and avoid any manual processes.

If I can get the information in red below with a script that would be what I’m looking for.

***@***.***

I looked at the Get-PnPSiteAnaliticsData and I’m not seeing the information that I’m looking for from the available data this cmdlet gives.

**Reply by Barbarur** [reply]
I think you can start with something like this:

Get-PnPTenantSite | Export-Csv "$Env:USERPROFILE\Documents\SiteCollectionReport.csv" -NoTypeInformation

**Comment by Oregonfrogger** [comment]
![SiteAdminInfo](https://github.com/pnp/powershell/assets/31940214/717570bd-26cf-4f53-a905-9001b0866911)

**Comment by Oregonfrogger** [comment]
Unfortunately, that cmdlet doesn’t have the following information:
Last Activity
Created by
Files viewed
Page views
Page visits
Files
